### PR TITLE
Native source code and a Podfile without a plugin

### PR DESCRIPTION
**Pull request by tdermendjiev**
## PR Checklist

- [x] The PR title follows our guidelines: https://github.com/NativeScript/NativeScript/blob/master/CONTRIBUTING.md#commit-messages.
- [x] There is an issue for the bug/feature this PR is for. To avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it.
- [x] You have signed the [CLA](http://www.nativescript.org/cla).
- [x] All existing tests are passing: https://github.com/NativeScript/nativescript-cli/blob/master/CONTRIBUTING.md#contribute-to-the-code-base
- [x] Tests for the changes are included.

## What is the current behavior?
Native source code files are included to the XCode project only when they are in plugins. Podfiles as well.

## What is the new behavior?
Podfile and source files added in App_Resources/iOS will be included as well.

Implements #3807 .

### CODE REVIEW

File: lib/services/cocoapods-service.ts [hunk 1]
```diff
@@ -52,39 +52,40 @@ export class CocoaPodsService implements ICocoaPodsService {
 		return podInstallResult;
 	}
 
-	public async applyPluginPodfileToProject(pluginData: IPluginData, projectData: IProjectData, nativeProjectPath: string): Promise<void> {
-		const pluginPodFilePath = this.getPluginPodfilePath(pluginData);
-		if (!this.$fs.exists(pluginPodFilePath)) {
+	public async applyPodfileToProject(moduleName: string, podfilePath: string, projectData: IProjectData, nativeProjectPath: string): Promise<void> {
+		if (!this.$fs.exists(podfilePath)) {
+			if (podfilePath === projectData.podfilePath) {
```
Comment (DimitarTachev) [inline lib/services/cocoapods-service.ts]: Why do we need this check?

File: lib/services/ios-project-service.ts [hunk 2]
```diff
@@ -771,20 +771,27 @@ We will now place an empty obsolete compatability white screen LauncScreen.xib f
 			this.$logger.trace(`Images to remove from xcode project: ${imagesToRemove.join(", ")}`);
 			_.each(imagesToRemove, image => project.removeResourceFile(path.join(this.getAppResourcesDestinationDirectoryPath(projectData), image)));
 
+			await this.prepareNativeSourceCode("src", path.join(projectData.appDirectoryPath, constants.APP_RESOURCES_FOLDER_NAME, this.getPlatformData(projectData).normalizedPlatformName, "src"), projectData);
```
Comment (DimitarTachev) [inline lib/services/ios-project-service.ts]: Shouldn't this `src` depend on the nsconfig?

Comment (DimitarTachev) [inline lib/services/ios-project-service.ts]: We could extract some "src" constants.

File: cliapp/platforms/ios/cliapp/Resources/Assets.xcassets/AppIcon.appiconset/Contents.json [hunk 3]
```diff
@@ -0,0 +1,98 @@
+{
```
Comment (rosen-vladimirov) [inline cliapp/platforms/ios/cliapp/Resources/Assets.xcassets/AppIcon.appiconset/Contents.json]: you should not commit this file

**Comment by Fatme** [comment]
run ci

**Comment by farfromrefug** [comment]
would this work for modules too? i mean i would love to be able to drop sources directly in my module folder (platforms/ios/src for example) and have them being picked up during module build without the need to have a XCode project or a android studio project.

**Comment by tdermendjiev** [comment]
@farfromrefug Yes, it works.

**Comment by farfromrefug** [comment]
@tdermendjiev awesome will try asap

**Pull request merged**